R/data.table.R

```diff
@@ -1739,22 +1739,48 @@ replace_dot_alias = function(e) {
         GForce = FALSE
       } else {
         # Apply GForce
+        noCall_noVars = function(expr) (!is.call(expr) || length(all.vars(expr, max.names=1L, unique=FALSE))==0L) && !dotN(expr)
+        .gshift_ok = function(q) {
+          q = match.call(shift, q)
+          noCall_noVars(q[["n"]]) &&
+            noCall_noVars(q[["fill"]]) &&
+            noCall_noVars(q[["type"]]) &&
+            is.null(q[["give.names"]])
```

Comment:
i.e. either `!give.names %chin% names(q) || <give.names manually supplied as NULL>` right?

Reply:
I would say `give.names` has to be missing since `gshift = function(x, n=1L, fill=NA, type=c("lag", "lead", "shift", "cyclic"))` and `shift = function(x, n = 1L, fill, type = c("lag", "lead", "shift", "cyclic"), give.names = FALSE)`.

Reply:
(resolved by suggested change in neighboring comment)